### PR TITLE
Fixup of initial jt-264-basics merge PR #282

### PR DESCRIPTION
**Pull request by laurentschoelens**
Fixing build issues of PR #282 : build is now successfull on jdk8/11/17 with maven profile `-P all`

Renaming all modules to JAXB Tools :: ...
Fixing build dependencies of samples / tests of jaxb-maven-plugin on jaxb2-basics

Also fixes https://github.com/highsource/jaxb2-basics/issues/130
Also fixes build issue (reflection access disallowed for URI class) on jdk17 (creating new internal class)

### CODE REVIEW

File: basics/runtime/src/main/java/org/jvnet/jaxb2_commons/lang/DefaultCopyStrategy.java [hunk 1]
```diff
@@ -267,6 +270,74 @@ protected Object copyInternal(ObjectLocator locator, Cloneable object) {
 			}
 		}
 	}
+	protected Object copyInternalJdk11(ObjectLocator locator, Cloneable object) {
```
Comment (mattrpav) [inline basics/runtime/src/main/java/org/jvnet/jaxb2_commons/lang/DefaultCopyStrategy.java:273]: This method has a lot of copy+paste from the other copyInternal.  Perhaps we can just add an argument to adjust behavior to the original method?

File: basics/tools/src/test/java/org/jvnet/jaxb2_commons/reflection/util/test/URIInternalTest.java
Comment (mattrpav) [inline basics/tools/src/test/java/org/jvnet/jaxb2_commons/reflection/util/test/URIInternalTest.java:1]: Why is this needed?

**Review by mattrpav — CHANGES_REQUESTED** [review]

File: basics/runtime/src/main/java/org/jvnet/jaxb2_commons/lang/DefaultCopyStrategy.java (hunk 1 above)
Comment (laurentschoelens) [reply basics/runtime/src/main/java/org/jvnet/jaxb2_commons/lang/DefaultCopyStrategy.java:273]: I didn't want to change existing behaviour for jdk8 / jdk11 which makes it works well.
The exception is only thrown with jdk17 since jdk16 has locked more aggressively the internal jdk classes.
This can be the default impl with jakarta-jaxb4 since baseline will be jdk11 (jaxb4 requires jdk11)

File: basics/tools/src/test/java/org/jvnet/jaxb2_commons/reflection/util/test/URIInternalTest.java
Comment (laurentschoelens) [reply basics/tools/src/test/java/org/jvnet/jaxb2_commons/reflection/util/test/URIInternalTest.java:1]: Needed to avoid setting extra argument in maven-surefire plugin to allow opens package of URI.
With jdk17, build failed because of the same problem as above : java.net.URI is no more accessible to reflection.
To test the ReflectionUtil, we should test it on a internal class rather than a JDK class

Comment (mattrpav) [reply basics/tools/src/test/java/org/jvnet/jaxb2_commons/reflection/util/test/URIInternalTest.java:1]: No prob, in the future let's make these types of changes a separate PR so we can back port across branches

**Review by mattrpav — CHANGES_REQUESTED** [review]
This looks good! Please add a unit test for the failure case (ie.. class w/ Cloneable interface, but no clone() method) and we'll be good to go!

**Comment by laurentschoelens** [comment]
> This looks good! Please add a unit test for the failure case (ie.. class w/ Cloneable interface, but no clone() method) and we'll be good to go!

Done, see org.jvnet.jaxb2_commons.lang.tests.DefaultCopyStrategyTest

**Review by mattrpav — APPROVED** [review]
LGTM
